File: Models/GrowingSpace.swift
import SwiftData

enum SpaceType: String, CaseIterable {
    case bed
    case container
    case row
    case other

    var displayName: String {
        switch self {
        case .bed: return "Bed"
        case .container: return "Container"
        case .row: return "Row"
        case .other: return "Other"
        }
    }
}

@Model
final class GrowingSpace {
    var id: UUID
    var name: String
    var notes: String?
    var spaceType: SpaceType?
    var width: Double?
    var length: Double?
    var createdAt: Date
    var updatedAt: Date

    @Relationship(deleteRule: .nullify, inverse: \Garden.growingSpaces)
    var garden: Garden?

    @Relationship(deleteRule: .cascade, inverse: \Occupancy.growingSpace)
    var occupancies: [Occupancy] = []

    @Relationship(deleteRule: .cascade, inverse: \PlannedPlanting.growingSpace)
    var plannedPlantings: [PlannedPlanting] = []

    init(name: String, notes: String? = nil, spaceType: SpaceType? = nil, width: Double? = nil, length: Double? = nil) {
        self.id = UUID()
        self.name = name
        self.notes = notes
        self.spaceType = spaceType
        self.width = width
        self.length = length
        self.createdAt = Date()
        self.updatedAt = Date()
    }
}

File: shared/src/commonMain/kotlin/com/soilandsupper/shared/domain/model/GrowingSpace.kt
package com.soilandsupper.shared.domain.model

data class GrowingSpace(
    val id: Long = 0,
    val name: String,
    val notes: String? = null,
    val spaceType: String? = null,
    val width: Double? = null,
    val length: Double? = null,
    val gardenId: Long? = null,
    val createdAt: Long = System.currentTimeMillis(),
    val updatedAt: Long = System.currentTimeMillis()
) {
    val area: Double?
        get() = if (width != null && length != null && width > 0 && length > 0) width * length else null
}

